lua/lean/init.lua

```diff
@@ -73,4 +70,36 @@ function lean.use_suggested_mappings(buffer_local)
   end
 end
 
+--- Is the current buffer a lean buffer?
+function lean.is_lean_buffer()
+  return vim.bo.ft == "lean" or vim.bo.ft == "lean3"
+end
+
+--- Return the current Lean search path.
+---
+--- Includes both the Lean core libraries as well as project-specific
+--- directories.
+function lean.current_search_paths()
+  local paths
+
+  if vim.bo.ft == "lean3" then
+    paths = require'lean.lean3'.__current_search_paths()
+  else
+    local root = vim.lsp.buf.list_workspace_folders()[1]
+    paths = vim.tbl_map(
+      function(path) return root .. '/' .. path end,
+      subprocess_check_output({"leanpkg", "print-paths"}, { cwd = root })
```

Comment:
This should only take the second line, and split it by `:`.  This is an example output:
```
$ leanpkg print-paths
././build:build/deps/Qq/./build
./.:build/deps/Qq/.
```

The directories with source files are `./.` and `build/deps/Qq/.`

Reply:
Aha, sorry, missed that comment. The first line is some unrelated information?

Reply:
I think the first line is the directories where the *.olean files are stored.

Reply:
Fixed I believe.